src/helpers/AvroToMetaModel.ts

```diff
@@ -0,0 +1,435 @@
+  return new AnyModel(
+    modelName,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel)
+  );
+}
+
+export function toBooleanModel(
+  avroSchemaModel: AvroSchema,
+  name: string
+): BooleanModel | undefined {
+  if (!avroSchemaModel.type?.includes('boolean')) {
+    return undefined;
+  }
+  return new BooleanModel(
+    name,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel)
+  );
+}
+export function toIntegerModel(
+  avroSchemaModel: AvroSchema,
+  name: string
+): IntegerModel | undefined {
+  if (
+    !avroSchemaModel.type?.includes('int') ||
+    !avroSchemaModel.type?.includes('long')
+  ) {
+    return undefined;
+  }
+  return new IntegerModel(
+    name,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel)
+  );
+}
+export function toFloatModel(
+  avroSchemaModel: AvroSchema,
+  name: string
+): FloatModel | undefined {
+  if (
+    !avroSchemaModel.type?.includes('float') ||
+    !avroSchemaModel.type?.includes('double')
+  ) {
+    return undefined;
+  }
+  return new FloatModel(
+    name,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel)
+  );
+}
+export function toStringModel(
+  avroSchemaModel: AvroSchema,
+  name: string
+): StringModel | undefined {
+  if (
+    !avroSchemaModel.type?.includes('string') ||
+    !avroSchemaModel.type?.includes('bytes')
+  ) {
+    return undefined;
+  }
+  return new StringModel(
+    name,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel)
+  );
+}
+export function toEnumModel(
+  avroSchemaModel: AvroSchema,
+  name: string
+): EnumModel | undefined {
+  if (
+    !avroSchemaModel.type?.includes('enum') ||
+    !Array.isArray(avroSchemaModel.symbols)
+  ) {
+    return undefined;
+  }
+  const enumValueToEnumValueModel = (enumValue: unknown): EnumValueModel => {
+    if (typeof enumValue !== 'string') {
+      return new EnumValueModel(JSON.stringify(enumValue), enumValue);
+    }
+    return new EnumValueModel(enumValue, enumValue);
+  };
+
+  const metaModel = new EnumModel(
+    name,
+    avroSchemaModel.originalInput,
+    getMetaModelOptions(avroSchemaModel),
+    []
+  );
+
+  if (avroSchemaModel.symbols) {
+    for (const enumValue of avroSchemaModel.symbols) {
+      metaModel.values.push(enumValueToEnumValueModel(enumValue));
+    }
+  }
+  return metaModel;
+}
+// checks if the typeof value can have any type
+function shouldBeAnyType(avroSchemaModel: AvroSchema): boolean {
+  // check the type array for the any type
+  const containsAllTypesButNotNull =
+    Array.isArray(avroSchemaModel.type) &&
+    avroSchemaModel.type.length >= 6 &&
+    !avroSchemaModel.type.includes('null');
+  const containsAllTypes =
+    (Array.isArray(avroSchemaModel.type) &&
+      avroSchemaModel.type.length === 7) ||
+    containsAllTypesButNotNull;
+  return containsAllTypesButNotNull || containsAllTypes;
```

Comment:
Seems like you are reusing incorrect implementation here from JSON Schema input 🤔